Add multi-ecosystem group to Dependabot config

parkerbxyz opened the pull request:
* Adds `multi-ecosystem-groups` with a new group named `minor-and-patch`, scheduled quarterly, to consolidate updates across ecosystems.
* Changes the `updates` section to use `patterns` for dependency matching and assigned updates to the new `multi-ecosystem-group`.

parkerbxyz committed:
Add multi-ecosystem group to Dependabot config

gr2m reviewed (approved):
still working on it? Looks like an improvement to what we have right now.

robaiken commented on .github/dependabot.yml:
```diff
@@ -21,8 +27,8 @@ updates:
-      interval: 'monthly'
+    patterns: ['*']
+    multi-ecosystem-groups: 'minor-and-patch'
     groups:
```
This group should also be removed

robaiken reviewed (commented):
The` commit-message` key needs to be added directly to the `multi-ecosystem-groups` section of your group definition, not nested within individual ecosystems

parkerbxyz committed:
Move commit-message config to multi-ecosystem group

Commit-message settings for minor-and-patch updates were relocated from the npm update section to the multi-ecosystem-groups.minor-and-patch section in dependabot.yml.

parkerbxyz committed:
Move `update-types` config for GitHub Actions

Removed the 'groups' section and moved 'update-types' directly under the GitHub Actions configuration.

parkerbxyz committed:
Merge branch 'main' into parkerbxyz/dependabot-multi-ecosystem-groups
